tango_simlib/tango_sim_generator.py

```diff
@@ -282,63 +297,82 @@ def initialize_dynamic_attributes(self):
+                attr = self._create_attribute(
+                    attribute_name, meta_data["data_type"], rw_type
+                )
+                if attr is None:
                     continue
-                elif str(d_format) == "IMAGE":
-                    self._not_added_attributes.append(attribute_name)
+
+                self._configure_attribute_default_properties(attr, meta_data)
+                self._add_dynamic_attribute(attr, rw_type)
+                MODULE_LOGGER.info("Added dynamic {} attribute".format(attribute_name))
+
+        def _add_dynamic_attribute(self, attribute, read_write_type):
+            if read_write_type in (AttrWriteType.READ, AttrWriteType.READ_WITH_WRITE):
+                self.add_attribute(attribute, r_meth=self.read_attributes)
+            elif read_write_type == AttrWriteType.WRITE:
+                self.add_attribute(attribute, w_meth=self.write_attributes)
+            elif read_write_type == AttrWriteType.READ_WRITE:
+                self.add_attribute(
+                    attribute, r_meth=self.read_attributes, w_meth=self.write_attributes
+                )
+
+        def _is_attribute_addable_dynamically(self, quantity_meta_data):
+            attr_dtype = quantity_meta_data["data_type"]
+            d_format = quantity_meta_data["data_format"]
+            if str(attr_dtype) == "DevEnum" or str(d_format) == "SPECTRUM":
+                return False
+            elif str(d_format) == "IMAGE":
+                self._not_added_attributes.append(quantity_meta_data["name"])
+                return False
+
+            return True
+
+        def _create_attribute(self, attribute_name, attr_dtype, rw_type):
+            attribute = None
+            # Add a try/except clause when creating an instance of Attr class
+            # as PyTango may raise an error when things go wrong.
+            try:
+                attribute = Attr(attribute_name, attr_dtype, rw_type)
+            except Exception as e:
+                self._not_added_attributes.append(attribute_name)
+                MODULE_LOGGER.info(
+                    "Attribute %s could not be added dynamically"
+                    " due to an error raised %s.",
+                    attribute_name,
+                    str(e),
+                )
+
+            return attribute
+
+        def _configure_attribute_default_properties(self, attribute, quantity_meta_data):
+            attribute_properties = UserDefaultAttrProp()
+            for prop, prop_value in quantity_meta_data.items():
+                # NB: Calling 'set_enum_labels' or setting the 'enum_labels' results
+                # in a error, and we do not need to do anyway as DevEnum attributes are
+                # handled by the `add_static_attribute` method.
+                if prop == "enum_labels":
                     continue
+                if hasattr(attribute_properties, prop):
+                    setattr(attribute_properties, prop, prop_value)
```

Comment:
Put this in a `try/except`. It silently fails when things go wrong, resulting in attributes not added to the device.